editor.planx.uk/src/@planx/components/MapAndLabel/Public/Context.tsx

```diff
@@ -119,7 +150,18 @@ export const MapAndLabelProvider: React.FC<MapAndLabelProviderProps> = (
   const isFeatureInvalid = (index: number) =>
     Boolean(get(formik.errors, ["schemaData", index]));
 
-  const addFeature = () => {
+  const addFeatureToMap = (geojson: GeoJSONChange) => {
+    resetErrors();
+    setFeatures(geojson["EPSG:3857"].features);
+    setActiveIndex((features && features?.length - 2) || activeIndex + 1);
```

Comment:
There's some ugly active tab index management going on! Very much need to revisit this / open to suggestions. 

In many instances, I want the active tab index to be the last in the list. Otherwise there can be some strange behavior, for example on staging:
- Add trees 1, 2, 3 on the map
- Navigate to Tab 1 (it's active)
- Add another tree (#4) to the map - but now tab 2 is active when we really want 4 to be active

Reply:
Yeah this is an issue in the List component which is avoided by only allowing a single active item at a time (it's not possible to add when there's an active item). That solution won't work here due to the map driven nature of the component. 

It's going to be a little tricky as `setFeatures()` is async, but we can take the length from `geojson["EPSG:3857"].features` reliably I believe.

Something like this might work?

```
const newFeatures = geojson["EPSG:3857"].features;
setFeatures(newFeatures);
setActiveIndex(newFeatures.length - 1)
```

I've not tried this so I might be off the mark a little here and treading over old ground you've already worked through.

An alternative might a `useEffect()` to catch changes to `features.length` and tie the update to `setActiveIndex` to that?